File: BotsSDK/korebot/src/main/java/kore/botssdk/fragment/header/BaseHeaderFragment.java
package kore.botssdk.fragment.header;

import android.widget.ImageView;

import androidx.fragment.app.Fragment;

import kore.botssdk.listener.ComposeFooterInterface;
import kore.botssdk.listener.InvokeGenericWebViewInterface;
import kore.botssdk.models.BrandingModel;

public abstract class BaseHeaderFragment extends Fragment {
    protected BrandingModel brandingModel;
    protected ComposeFooterInterface composeFooterInterface;
    protected InvokeGenericWebViewInterface invokeGenericWebViewInterface;
    public void setComposeFooterInterface(ComposeFooterInterface composeFooterInterface){
        this.composeFooterInterface =composeFooterInterface;
    }

    public void setInvokeGenericWebViewInterface(InvokeGenericWebViewInterface invokeGenericWebViewInterface){
        this.invokeGenericWebViewInterface = invokeGenericWebViewInterface;
    }

    public abstract void setBrandingDetails(BrandingModel brandingModel);

    public abstract ImageView getMinimize();

}

File: BotSdk/ui/src/main/java/com/kore/ui/botchat/fragment/BaseHeaderFragment.kt
package com.kore.ui.botchat.fragment

import androidx.fragment.app.Fragment
import com.kore.common.event.UserActionEvent
import com.kore.network.api.responsemodels.branding.BrandingHeaderModel

abstract class BaseHeaderFragment : Fragment() {
    abstract fun setActionEvent(onActionEvent: (event: UserActionEvent) -> Unit)
    abstract fun setBrandingDetails(brandingModel: BrandingHeaderModel?)
}